### PR TITLE
Use route handle for common page data

### PR DESCRIPTION
**Pull request by sebastien-comeau**
As per title.

### CODE REVIEW

File: frontend/app/routes/_gcweb-app.$.tsx [hunk 1]
```diff
@@ -7,15 +7,14 @@ import { getTypedI18nNamespaces } from '~/utils/locale-utils';
...
-  );
+export const handle = {
+  i18nNamespaces,
+  pageIdentifier: 'CDCP-0404',
+  pageTitleI18nKey: 'gcweb:not-found.page-title',
+} as const satisfies RouteHandleData;
+
+export const loader = async () => {
+  return json(null, { status: 404 });
 };
```
Comment (gregory-j-baker) [inline frontend/app/routes/_gcweb-app.$.tsx:18]:
```suggestion
export async function loader({ request }: LoaderFunctionArgs) {
  return json(null, { status: 404 });
};
```

File: frontend/app/utils/locale-utils.ts [hunk 2]
```diff
@@ -1,8 +1,9 @@
-import { type FlatNamespace, type Namespace, createInstance } from 'i18next';
+import { type FlatNamespace, createInstance } from 'i18next';
 import I18nextBrowserLanguageDetector from 'i18next-browser-languagedetector';
 import I18NextHttpBackend from 'i18next-http-backend';
 import { initReactI18next } from 'react-i18next';
 
+import { i18nNamespacesSchema } from './route-utils';
```
Comment (gregory-j-baker) [inline frontend/app/utils/locale-utils.ts:6]:
```suggestion
import { i18nNamespacesSchema } from '~/utils/route-utils';
```

File: frontend/app/utils/locale-utils.ts [hunk 3]
```diff
@@ -52,6 +53,7 @@ export async function initI18n(namespaces: Array<string>) {
     .init({
       appendNamespaceToMissingKey: true,
       debug: I18NEXT_DEBUG,
+      defaultNS: 'common',
```
Comment (gregory-j-baker) [inline frontend/app/utils/locale-utils.ts:56]: I feel like this should either be `defaultNS: false` or `defaultNS: 'gcweb'`. Preferrably `false`.

Eventually we will git rid of the `common` namespace and replace it with something that is a little more descriptive.

**Review by gregory-j-baker — CHANGES_REQUESTED** [review]

File: frontend/app/routes/_gcweb-app.$.tsx (hunk 1 above)
Comment (sebastien-comeau) [reply frontend/app/routes/_gcweb-app.$.tsx]: We're using arrow function in a coulpe of places and I fixed them. `request` object is not used in this case and it's why I removed it. I added it as you're suggested.

File: frontend/app/utils/locale-utils.ts (hunk 3 above)
Comment (sebastien-comeau) [reply frontend/app/utils/locale-utils.ts]: I set `defaultNS: false`

File: frontend/app/routes/_gcweb-app.$.tsx (hunk 1 above)
Comment (gregory-j-baker) [reply frontend/app/routes/_gcweb-app.$.tsx]: It's not terrible to remove it, but I find it helps when debugging issues to have it available. The eslint rule that barfs when a function parameter is not used is turned off for the project (as it should be, imo) so having the parameter there isn't a problem.

**Review by gregory-j-baker — APPROVED** [review]
Well done my friend. Well done.

**Pull request merged**